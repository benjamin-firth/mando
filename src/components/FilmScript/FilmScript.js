import React from 'react';
import PropTypes from 'prop-types';

export const FilmScript = ({ crawl, filmTitle }) => {
  return (
    <div className='FilmScript'>
      <div className="fade"></div>
      <section className="star-wars">
        <div className="crawl">
          <div className="title">
              <h4>
                {filmTitle}
              </h4>
          </div>
          <p className='opening-text'>
            {crawl}
          </p>
        </div>
      </section>
    </div>
  )
};

FilmScript.propTypes = {
  crawl: PropTypes.string,
  filmTitle: PropTypes.string
}

export default FilmScript;
