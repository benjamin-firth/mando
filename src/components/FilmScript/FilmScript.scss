.FilmScript {
  height: 100%;
  overflow: hidden;
  width: 50%;
  // margin: auto;
  max-height: 79vh;
}

.fade {
  position: relative;
  min-height: 35vh;
  top: 0px;
  z-index: 1;
}

.star-wars {
  color: #feda4a;
  display: flex;
  justify-content: center;
  position: relative;
  height: 800px;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
}

.opening-text {
  color: #feda4a;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h4 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
	0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
	100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}