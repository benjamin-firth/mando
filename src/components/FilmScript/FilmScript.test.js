import React from 'react';
import { FilmScript } from './FilmScript';
import { shallow } from 'enzyme';

describe('FilmScript', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FilmScript />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})