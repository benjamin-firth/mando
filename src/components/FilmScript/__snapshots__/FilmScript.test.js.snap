// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmScript should match the snapshot 1`] = `
<div
  className="FilmScript"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <h4 />
      </div>
      <p
        className="opening-text"
      />
    </div>
  </section>
</div>
`;
